docs/import/list_ee_assets.js

```diff
@@ -1,41 +1,111 @@
+ * Cache for the results of getStatesAssetsFromEe for each state.
+ * @type {Map<string, Promise<Map<string, string>>>}
+ */
+const stateAssetPromises = new Map();
+
+/**
+ * Gets all assets in ee directories corresponding to given states. Caches
+ * results. Marks assets with their type and whether or not they should be
+ * disabled when put into a select. Here, disabling any assets that aren't
+ * feature collections.
+ * @param {Array<string>} states
+ * @return {Map<string, Promise<Map<string, {disabled: boolean}>>>}
  */
 function getStatesAssetsFromEe(states) {
-  const promises = [];
+  const toReturn = new Map();
   for (const state of states) {
-    const dir = legacyStateDir + '/' + state;
-    promises.push(listEeAssets(dir).then((result) => [state, result]));
+    const maybeStatePromise = stateAssetPromises.get(state);
+    if (maybeStatePromise) {
+      toReturn.set(state, maybeStatePromise);
+      continue;
+    }
+    const statePromise =
+        listEeAssets(legacyStateDir + '/' + state).then((result) => {
+          const assetMap = new Map();
+          for (const [asset, type] of result) {
+            assetMap.set(
+                asset, {disabled: type !== LayerType.FEATURE_COLLECTION});
+          }
+          return assetMap;
+        });
+    stateAssetPromises.set(state, statePromise);
+    toReturn.set(state, statePromise);
   }
-  return Promise.all(promises);
+  return toReturn;
 }
 
+/**
+ * Cache for the results of getDisasterAssetsFromEe for each disaster
+ * @type {Map<string, Promise<Map<string, {type: LayerType, disabled:
+ *     boolean}>>>}
+ */
 const disasterAssetPromises = new Map();
 
 /**
  * Gets all assets for the given disaster. Assumes an ee folder has already
- * been created for this disaster.
+ * been created for this disaster. Marks assets with their type and whether or
+ * not they should be disabled when put into a select. Here, disabling any
+ * feature collections that have a null-looking (see comment below) geometry.
  *
- * Deduplicates requests, so retrying before a fetch completes won't start a new
- * fetch.
+ * De-duplicates requests, so retrying before a fetch completes won't start a
+ * new fetch.
  * @param {string} disaster disaster in the form name-year
- * @return {Promise<Map<string, string>>} Returns a promise containing the map
- * of asset path to type for the given disaster.
+ * @return {Promise<Map<string, {type: number, disabled: boolean}>>} Returns
+ *     a promise containing the map of asset to info for the given disaster.
  */
 function getDisasterAssetsFromEe(disaster) {
   const maybePromise = disasterAssetPromises.get(disaster);
   if (maybePromise) {
     return maybePromise;
   }
-  const result = listEeAssets(eeLegacyPathPrefix + disaster);
+  const result =
+      listEeAssets(eeLegacyPathPrefix + disaster)
+          .then((assetMap) => {
+            const shouldDisable = [];
+            for (const asset of Array.from(assetMap.keys())) {
```

Comment:
again, this could be `for (const [asset, type] of Array.from(assetMap.entries()))`